feat(pages): create new theme page examples

IgnacioBecerra opened the pull request:
### Related Ticket(s)
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6732

### Description
Create Dark and White theme page examples.

### Changelog

**New**

- Dark and White theme example pages added

RichKummer commented:
Working through white theme, but here is some feedback so far:

For the initial page:
1. "Page template" should be "Page examples"
2. "White theme page" should be "White theme example page".
3. "Gray 100 theme page" should be "Gray 100 theme example page".
<img width="1781" alt="Screen Shot 2021-10-01 at 3 11 48 PM" src="https://user-images.githubusercontent.com/15643582/135681741-7f0f4b83-9c9b-416c-80d2-28fe3ad3bb37.png">

White theme example page:
**Lead Space**
1. In the `lead space`, the "See components in action" CTA should use the arrow-down icon, not the arrow-right icon.
2. Right now, the "See components in action" button takes you instantly to the first section. Can this be smooth-scrolled to the destination?
3. The "See alternate template" button should navigate the user to the Gray100 theme example page.
<img width="1715" alt="Screen Shot 2021-10-01 at 4 22 30 PM" src="https://user-images.githubusercontent.com/15643582/135681910-01ceca24-2a06-4d43-9ef7-75fe58668454.png">

**Vertical ToC** 
1. The vertical ToC should be on the grid.
2. Missing items for "Content group banner", "Content group cards" and "Logo grid".
<img width="1719" alt="Screen Shot 2021-10-01 at 4 23 43 PM" src="https://user-images.githubusercontent.com/15643582/135682171-a5cac6be-2dad-4147-bbfc-cc61cbea2624.png">

**Content Block - Simple**
1. The "Learn about Content block simple" card should navigate the user to [https://www.ibm.com/standards/web/carbon-for-ibm-dotcom/components/content-block-simple](https://www.ibm.com/standards/web/carbon-for-ibm-dotcom/components/content-block-simple). It should still open in a new tab as it does now.
2. It does not look like Content block simple is on the grid. The CTA card is also not on the grid.
3. Card should use `expressive-heading-01` type token.
4. As the user scrolls down the page, the card component should animate in after the body text. Right now it appears on-load and never animates in.
<img width="1719" alt="Screen Shot 2021-10-01 at 4 23 43 PM 1" src="https://user-images.githubusercontent.com/15643582/135682353-ac9b80de-9ea8-4802-a9d4-5d2c6b98870e.png">

**Content block media** 
1. The image link under "Introduction to the media" is not rendering (it appears as a broken link.
2. The in-line text link "design specs" should open the link in a new tab.
3. Same with the "Storybook" link at the bottom of the "Scroll animations" paragraph.
4. For the card in the section, please update the "lorem ipsum dolor sit ametttt" to "Content block media". Have the link open this page in a new tab: [https://www.ibm.com/standards/carbon/components/content-block-media](https://www.ibm.com/standards/carbon/components/content-block-media)
5. Card title should use `expressive-heading-01`. 
6. Aside elements should align the text to the grid with 16px overhang from the horizontal rule.
7. As the user scrolls down the page, the aside element list should animate in after the body copy has animated in. Right now it appears on-load and never animates in.
<img width="1726" alt="Screen Shot 2021-10-01 at 4 29 27 PM" src="https://user-images.githubusercontent.com/15643582/135682789-8ac23dfd-ded1-4402-b39a-c9131bafd8f0.png">

**Callout quote**
1. In the `Callout quote`, the CTA link should be on the grid.
2. The CTA should also use the `inverse-link` color token. 
3. The "See more about the Callout quote" CTA should open in a new tab.
4. The Callout quote background should extend 16 pixels from the grid on either side - right now it extends much further.
<img width="1693" alt="Screen Shot 2021-10-01 at 3 19 13 PM" src="https://user-images.githubusercontent.com/15643582/135683093-0af52c8f-43f9-46a7-b8f7-933a5e9ac634.png">

**Content group pictograms**
1. The "Content group pictograms" and "IBM Design language pictogram library" CTAs should both open in a new tab.
2. The second card's body copy should end with a period.
3. On breakpoints smaller than `xlg`, the content of this section goes off the grid.
<img width="949" alt="Screen Shot 2021-10-01 at 4 17 31 PM" src="https://user-images.githubusercontent.com/15643582/135683237-aecc275d-d3a9-42a0-81e2-3578e5e6e6d0.png">

**Content group cards**
1. The heading, body, and cards do not appear to be aligned to the grid.
2. As the user scrolls down the page, the card components should animate in after the body copy has animated. Right now, the cards appear on-load and never animate as the user scrolls. The individual logo grid items animate in separately - these cards should follow that pattern.
<img width="1580" alt="Screen Shot 2021-10-01 at 4 36 36 PM" src="https://user-images.githubusercontent.com/15643582/135683551-895aea30-7aa4-4ef4-b457-ff0e713d45bb.png">

**Logo grid**
1. The heading and the logo cards should be updated to align with the grid. This includes smaller screen size breakpoints.
<img width="675" alt="Screen Shot 2021-10-01 at 4 40 03 PM" src="https://user-images.githubusercontent.com/15643582/135683723-59a55bbd-fc06-4e62-9dbb-3f20ef6f25ca.png">

<img width="1584" alt="Screen Shot 2021-10-01 at 4 39 36 PM" src="https://user-images.githubusercontent.com/15643582/135683720-594834d4-7b8c-4570-9096-c5625cf20bc8.png">

**Content group banner**
1. The "Content group banner" CTA should open in a new tab.
2. On `md` breakpoint, the content of this component hangs off the grid.
<img width="674" alt="Screen Shot 2021-10-01 at 4 42 17 PM" src="https://user-images.githubusercontent.com/15643582/135684028-ca58dd5e-79bd-441e-9463-45f739aceab7.png">

**Card section**
1. All cards in this component should navigate the user to this page in a new tab: [https://www.ibm.com/standards/carbon/components/card-section](https://www.ibm.com/standards/carbon/components/card-section)
2. The "Card section" title should be aligned to the grid on the left-hand side.
3. The cards should also align to the grid.
<img width="1586" alt="Screen Shot 2021-10-01 at 4 48 14 PM" src="https://user-images.githubusercontent.com/15643582/135684512-0ab5aa33-9d2d-46c0-9045-24b4c46aaf62.png">


**Card section carousel**
1. "See all components" CTA should open link in a new tab.
2. The "Card section carousel" title, the body copy, and CTA underneath it should align to the grid on the left-hand side. The cards should also align to the grid.
3. When I click on the carousel controls, the cards seem to pick up the `fade-in-up` animation style. This should only happen when the user scrolls the component into view. When the user clicks on the carousel controls, it should only animate side-to-side.
<img width="1509" alt="Screen Shot 2021-10-01 at 4 44 10 PM" src="https://user-images.githubusercontent.com/15643582/135684073-58fb8e39-93e5-436b-823e-86ac7ef1a7c6.png">

**CTA section**
1. The "See alternate template" CTA seems to be a broken link. Clicking it should open the Gray 100 theme example page.
2. The "Carbon for IBM.com" CTA should open this page in a new tab: [https://www.ibm.com/standards/carbon/](https://www.ibm.com/standards/carbon/)
3. The "Find a Partner" CTA should read as "More on CTA section". It should also open the link in a new tab.
4. The "More on CTA section" CTA under Usage should also open in a new tab.
5. This section should align to the grid.
6. As the user scrolls down this section, the individual elements should animate. Right now, only the heading and paragraph animate into view. The CTA buttons and subsections should also animate as the user scrolls further.
<img width="1585" alt="Screen Shot 2021-10-01 at 4 46 46 PM" src="https://user-images.githubusercontent.com/15643582/135684403-48cd0982-b872-4472-b793-c1b0100fe1c2.png">

RichKummer commented:
Gray 100 theme example page:

**Horizontal ToC**
1. The ToC overhangs the grid. Seems like the first text link is aligned right at the left-hand corner of the page.
<img width="1583" alt="Screen Shot 2021-10-01 at 4 58 00 PM" src="https://user-images.githubusercontent.com/15643582/135685484-5ec9c264-34f9-4add-804b-dfac7429f4ba.png">

**Card section images**
1. "Card section images" title overhangs off the grid - this should be aligned.
2. The text of the cards should be aligned with the columns. Right now, the second and fifth cards seem to be the only ones aligned on larger screen sizes. On `md`, it is only the right-hand cards that have their text aligned to the grid.
3. Clicking any of the cards should open the link in a new tab.
<img width="1575" alt="Screen Shot 2021-10-01 at 5 01 34 PM" src="https://user-images.githubusercontent.com/15643582/135686404-9a777309-9668-440a-9358-48482c4a656c.png">


**Feature card - large**
1. Right-hand edge of card should overhang by 24 px instead of extending to right edge of the screen on `max`. 
2. Clicking the Feature card - large should open the link in a new tab.
<img width="1581" alt="Screen Shot 2021-10-01 at 5 04 05 PM" src="https://user-images.githubusercontent.com/15643582/135685993-934cc8c0-8bc2-4b67-b016-9c724f654ffa.png">

**Content block segmented**
1. The content for this component is off the grid. Both the main content and the Link list should have their text aligned to the grid. 
2. The first item under Link list should be "Content block segmented design specs". 
3. Link list items do not appear to be working. The first should link to this page in a new tab: https://ibm.ent.box.com/folder/99730444060?s=pkascpokk7ae4t3ixis98hmw4e5y1v56. The second should link to this page in a new tab: https://ibm.ent.box.com/folder/99730444060?s=pkascpokk7ae4t3ixis98hmw4e5y1v56.
4. The aside link list should animate in after the expressive-paragraph-01 copy has animated in. Right now, the link list appears on-load and does not animate at all.
5. The paragraph copy in the subsections does not currently animate. Each section body copy should animate in after the heading animates in, one after the other. 
6.The card should also animate in once the user has scrolled down enough. Right now it does not animate at all.
<img width="1584" alt="Screen Shot 2021-10-01 at 5 10 53 PM" src="https://user-images.githubusercontent.com/15643582/135686555-34ebce62-ccd9-468e-85ff-c0fe56936eca.png">

**Content block horizontal**
1. The "Learn more" link should open in a new tab. "See design specs" link is working as intended.
2. The "See Web Component Storybook" CTA link should open in a new tab. "See React Storybook" link is working as intended.
3. Content for this section is off the grid:
<img width="1585" alt="Screen Shot 2021-10-01 at 5 13 37 PM" src="https://user-images.githubusercontent.com/15643582/135686887-2583b455-e910-4b0c-bb34-8b395028357a.png">
4. The content sections do not animate in (although the heading does). Can each section animate in as the user scrolls down the page?

**Tabs extended media**
1. "Tabs extended with media" title should be aligned with the grid. Currently the title begins at the left-hand side of the window. 
2. Double check the links for each tab in the Box note:
3. "Explore Tabs extended with media" should open this link in a new tab: https://www.ibm.com/standards/carbon/components/tabs-extended-media
4. "Learn about images" should open https://www.ibm.com/standards/carbon/components/video in a new tab.
5. "Explore Tabs extended with media" link should open link in a new tab.
6. "Learn about video" link should open in a new tab.
7. "Explore Tabs extended" link should open in a new tab.
8. The first time the user clicks on a tab, the content animates in with `fade-in-up`, as if the user were scrolling into view. Then the animation no longer applies the next time the user clicks that tab - this feels like a bug. Let's remove the animation when the user clicks a tab.

**Link list** 
1. Each link in the link list should open in a new tab.
2. The title for this section should read "Link list section can display links to other resources"

**Card section carousel**
1. The section on the left with the title, body, and CTA overhang the grid. The cards also should have their text aligned with the grid. 
2. "See all components" CTA should open link in a new tab.
3. Like the White theme version, when I click on the carousel controls, the cards seem to pick up the fade-in-up animation style. This should only happen when the user scrolls the component into view. When the user clicks on the carousel controls, it should only animate side-to-side.
<img width="1581" alt="Screen Shot 2021-10-01 at 5 23 05 PM" src="https://user-images.githubusercontent.com/15643582/135687833-00a745f0-ecd9-49bd-bc35-b5a37178c5f3.png">

**CTA section**
1. See comments on White theme version, except the CTA section on Gray 100 theme should have a link list instead of the subsections (see the design):
<img width="797" alt="Screen Shot 2021-10-01 at 5 26 36 PM" src="https://user-images.githubusercontent.com/15643582/135688036-21e4b996-a4ee-41c5-b5ae-8a6e2ca9a373.png">

jeffchew commented:
@IgnacioBecerra just checking status of this. Can you also create e2e tests for the new pages? (might have forgotten to check if they were added for NextJS as well).

IgnacioBecerra commented:
@RichKummer addressed most feedback in the last commit, but I'll reach out to you about the grid comments since it's a tricky issue

@jeffchew had been working on the above comments, but I made sure to add the e2e tests in the last commit as well. I'll open a PR to do the same for NextJS too 👍

jeffchew commented:
Thanks @IgnacioBecerra ! 

It looks like the tests are throwing some 404s: 
![image](https://user-images.githubusercontent.com/1641214/136231995-e18abd2b-6f35-4c6f-a496-6a29c4f7fe31.png)
